source/wrangling.md

```diff
@@ -2005,138 +1349,259 @@ An alternative to aggregating on a dataframe
-> `map` functions paired with base R statistical summary functions
-> also return `NA` values when we apply them to columns that 
-> contain `NA` values. \index{missing data}
-> 
-> To avoid this, again we need to add the argument `na.rm = TRUE`.
-> When we use this with `map`, we do this by adding a `,` 
-> and then `na.rm = TRUE` after specifying the function, as illustrated below:
-> 
-> ``` {r}
-> region_lang_na |>
->   select(mother_tongue:lang_known) |>
->   map_dfr(max, na.rm = TRUE)
-> ```
-
-The `map` functions are generally quite useful for solving many problems 
-involving repeatedly applying functions in R. 
-Additionally, their use is not limited to columns of a data frame;
-`map` family functions can be used to apply functions to elements of a vector,
-or a list, and even to lists of (nested!) data frames.
-To learn more about the `map` functions, see the additional resources
-section at the end of this chapter. -->
+(pandas-assign)=
+## Using `.assign` to modify or add columns
+
++++
+
+### Using `.assign` to create a new data frame
+
+If you would like to use the summary statistics that you computed in a further analysis, it can be handy
+to store them in a data frame that you can use later.
+We can use `.assign` as mentioned in along with proper summary functions to create a aggregated column.
+
+For example, say we wanted to keep the minimum and maximum values in a data frame, we can (1) create a new, empty
+data frame and (2) use `.assign` to assign values to that data frame.
+```
+region_lang_min_max = pd.DataFrame()  # empty data frame
+region_lang_min_max.assign(
+  most_at_home_min = region_lang["most_at_home"].min(),
+  most_at_home_max = region_lang["most_at_home"].max()
+)
+region_lang_min_max
+```
+
+### Using `.assign` to modify columns
+
+```{index} pandas.DataFrame; []
+```
+
+In the section on {ref}`str-split`,
+when we first read in the `"region_lang_top5_cities_messy.csv"` data,
+all of the variables were "object" data types.
+During the tidying process,
+we used the `pandas.to_numeric` function
+to convert the `most_at_home` and `most_at_work` columns
+to the desired integer (i.e., numeric class) data types and then used `[]` to overwrite columns.
+But suppose we didn't use the `[]`,
+and needed to modify the columns some other way.
+Below we create such a situation
+so that we can demonstrate how to use `.assign`
+to change the column types of a data frame.
+`.assign` is a useful function to modify or create new data frame columns.
+
+```{code-cell} ipython3
+lang_messy = pd.read_csv("data/region_lang_top5_cities_messy.csv")
+lang_messy_longer = lang_messy.melt(
+    id_vars=["category", "language"],
+    var_name="region",
+    value_name="value",
+)
+tidy_lang_obj = (
+    pd.concat(
+        (lang_messy_longer, lang_messy_longer["value"].str.split("/", expand=True)),
+        axis=1,
+    )
+    .rename(columns={0: "most_at_home", 1: "most_at_work"})
+    .drop(columns=["value"])
+)
```

Comment:
I am not sure this is critical to repeat. We could just pick up with `official_langs`?